pkg/pluginmanager/manager.go

```diff
@@ -737,26 +744,71 @@ func installOrUpgradePlugin(p *discovery.Discovered, version string, installTest
 	}
 
 	// Log message based on different installation conditions
-	logPluginInstallationMessage(p, version, plugin != nil, isPluginAlreadyInstalled)
+	installingMsg, installedMsg, errMsg := getPluginInstallationMessage(p, version, plugin != nil, isPluginAlreadyInstalled)
 
-	if plugin == nil {
-		binary, err := fetchAndVerifyPlugin(p, version)
-		if err != nil {
-			return err
+	var spinnerErr, pluginErr error
+	var sow component.OutputWriterSpinner
+
+	// Initialize the spinner if the spinner is allowed
+	if component.IsTTYEnabled() {
+		// Set spinner options
+		spinnerOptions := component.OutputWriterSpinnerOptions{
+			SpinnerOptions: []component.OutputWriterSpinnerOption{
+				component.WithSpinnerFinalText(installedMsg, log.LogTypeINFO),
+			},
+		}
+		// Initialize the spinner
+		sow, spinnerErr = component.NewOutputWriterSpinnerWithSpinnerOptions(os.Stderr, component.TableOutputType, installingMsg, true, spinnerOptions)
+		if spinnerErr != nil {
+			log.V(6).Infof("Unable to initialize spinner: %v", spinnerErr.Error())
+			log.Info(installingMsg)
 		}
+		if sow != nil {
+			defer sow.StopSpinner()
 
-		plugin, err = installAndDescribePlugin(p, version, binary)
-		if err != nil {
-			return err
+			// Create a channel to receive OS signals
+			signalChannel := make(chan os.Signal, 1)
+
+			// Register the channel to receive interrupt signals (e.g., Ctrl+C)
+			signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
+
+			// Start a goroutine that listens for interrupt signals
+			go func() {
+				// Wait for an interrupt signal
+				<-signalChannel
+
+				// Handle the interrupt signal by setting the final text and stopping the spinner
+				sow.SetFinalText(errMsg, log.LogTypeERROR)
+				sow.StopSpinner()
+
+				// Exit the program with exit code 130 (interrupted)
+				os.Exit(130)
```

Comment:
nit: It would look less mysterious if we use 128 + syscall.SIGINT instead, or, alternatively, use a clearer comment explaining how 130 came about.

Reply:
updated.